rfcs/063-catalogue_pipeline_lambdas/README.md

```diff
@@ -0,0 +1,139 @@
+
+Potential issues may arise around this change in exectution model, specifically:
+
+### Concurrency
+
+With ECS concurrency can be limited by the number of running tasks which provides a _rough_ restriction on the number of messages processed at once. With AWS Lambda concurrency can be specified exactly with ["reserved concurrency"](https://docs.aws.amazon.com/lambda/latest/dg/configuration-concurrency.html), and in combination with [batch size](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#sqs-polling-behavior) can tightly control the number of messages a Lambda processes.
+
+As we are moving from one model to another, we will need to calibrate concurrency for each service by estimation and experiment. 
+
+Concurrency is of particular concern where we are making network requests to external services that may not be configured to meet demand, for example Elasticsearch or RDS databases.
+
+### Batching
+
+Batching describes the number of messages handed to a Lambda to process in one invocation. For the catalogue pipeline we will be invoking Lambdas from non-FIFO SQS messages, and this has [specific restrictions around batch size](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html). 
+
+Batch size is dictated by a combination of:
+
+- Configured maximum batch size, up to [a maximum of 10,000 for non-FIFO queues](https://repost.aws/knowledge-center/lambda-sqs-scaling)
+- Maximum payload size, up to [128kb for asynchronous invocation](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html)
+- Length of batch window, (how long to wait before passing on a batch), [up to 5 minutes](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html).
+
+Along with concurrency batching dictates how much work is processed at one time. 
+
+A specific concern around batch size: 
+
+> The maximum number of records in each batch that Lambda pulls from your stream or queue and sends to your function. Lambda passes all of the records in the batch to the function in a single call, up to the payload limit for synchronous invocation 
+  (6 MB). The maximum batchsize for lambda is 10 000. This is far less than the batch size that the batcher is currently able to process (up to 120000 `collectionPath`). This could mean fewer nodes are being matched in a batch, reducing the 
+  beneficial effect of the batcher on the relation_embedder load. As of 2024-10-23 there are 271791 documents with a collectionPath in the merged index, ie. as many messages that the batcher needs to process as part of a full reindex
+
```

Comment:
It might be worth mentioning here that even in the current system,  `flushInterval` and `maxBatchSize` are set to try to balance between "wait until you have loads of messages" and "don't wait forever", so it does not always process 120000.

During a reindex, I'm pretty sure there are more than three invocations of the batcher.